hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestRecursiveAclWithFSOBucket.java

```diff
@@ -0,0 +1,333 @@
+     *         b1          b2             b3      file1
+     *       -----       ------           -----
+     *       |    |      |    |          |    |
+     *      c1   c2     d1   d2          e1   e2
+     *       |    |      |    |           |    |
+     *       f1   f2     f3  --------     f5   f6
+     *                      |        |
+     *                    d21        file2
+     *                     |
+     *                     f4
+     *
+     *     Test Case 1 :
+     *     Remove delete acl from file File2
+     *     Try deleting b2
+     *
+     *     Test case 2:
+     *     Remove delete acl fro dir c2
+     *     Try deleting b1
+     *
+     *     Test case 3
+     *     try deleting b3
+     */
+
+    String keyc1 = "a/b1/c1/f1";
+    String keyc2 = "a/b1/c2/f2";
+    String keyd1 = "a/b2/d1/f3";
+    String keyd21 = "a/b2/d2/d21/f4";
+    String keye1 = "/a/b3/e1/f5";
+    String keye2 = "/a/b3/e2/f6";
+
+    String file1 = "a/" + "file" + RandomStringUtils.randomNumeric(5);
+    String file2 = "a/b2/d2/" + "file" + RandomStringUtils.randomNumeric(5);
+
+    keys.add(keyc1);
+    keys.add(keyc2);
+    keys.add(keyd1);
+    keys.add(keyd21);
+    keys.add(keye1);
+    keys.add(keye2);
+    keys.add(file1);
+    keys.add(file2);
+
+    createKeys(objectStore, ozoneBucket, keys);
+
+    // Test case 1
+    // Remove acls from file2
+    // Delete/Rename on directory a/b2 should throw permission denied
+    // (since file2 is a child)
+    removeAclsFromKey(objectStore, ozoneBucket, file2);
+    OzoneObj ozoneObj;
+    List<OzoneAcl> aclList1;
+
+    UserGroupInformation.setLoginUser(user2);
+    client = cluster.getClient();
+    objectStore = client.getObjectStore();
+    volume = objectStore.getVolume("volume1");
+    ozoneBucket = volume.getBucket("bucket1");
+
+    // perform  delete
+    try {
+      ozoneBucket.deleteDirectory("a/b2", true);
+      Assert.fail("Should throw permission denied !");
+    } catch (OMException ome) {
+      // expect permission error
+      GenericTestUtils.assertExceptionContains("PERMISSION_DENIED", ome);
+    }
+
+    // perform rename
+    try {
+      ozoneBucket.renameKey("a/b2", "a/b2_renamed");
+      Assert.fail("Should throw permission denied !");
+    } catch (OMException ome) {
+      // expect permission error
+      GenericTestUtils.assertExceptionContains("PERMISSION_DENIED", ome);
+    }
+
+    // Test case 2
+    // Remove acl from directory c2, delete/rename a/b1 should throw
+    // permission denied since c2 is a subdirectory
+
+    UserGroupInformation.setLoginUser(user1);
+    removeAclsFromKey(objectStore, ozoneBucket, "a/b1/c2");
+
+    UserGroupInformation.setLoginUser(user2);
+    // perform  delete
+    try {
+      ozoneBucket.deleteDirectory("a/b1", true);
+      Assert.fail("Should throw permission denied !");
+    } catch (OMException ome) {
+      // expect permission error
+      GenericTestUtils.assertExceptionContains("PERMISSION_DENIED", ome);
```

Comment:
Please assert result code.

assertRequals("Permission check failed", ResultCodes.PERMISSION_DENIED, ome.getResult());

Please do this for the catch blocks, below  the test code as well..